### PR TITLE
Add ruff lint and style checks

### PR DESCRIPTION
**Pull request by jonchang**
# Pull Request

## Description
This implements the ruff formatter and linter for automatic style checks using github actions.

## Related Issues
See https://github.com/CDCgov/IDWA/blob/main/open_practices.md#recommended-practices, configuration for linters.

## Additional Notes
Line length 118 was chosen because (at least last time I checked) it is the maximum width that the github diff viewer has before soft wrapping.

## Checklist
Please review and complete the following checklist before submitting your pull request:

- [x] I have ensured that the pull request is of a manageable size, allowing it to be reviewed within a single session.
- [x] I have reviewed my changes to ensure they are clear, concise, and well-documented.
- [ ] I have updated the documentation, if applicable.
- [ ] I have added or updated test cases to cover my changes, if applicable.
- [ ] I have minimized the number of reviewers to include only those essential for the review.
- [ ] I have notified teammates in the review thread to build awareness.

## Checklist for Reviewers
Please review and complete the following checklist during the review process:

- [ ] The code follows best practices and conventions.
- [ ] The changes implement the desired functionality or fix the reported issue.
- [ ] The tests cover the new changes and pass successfully.
- [ ] Any potential edge cases or error scenarios have been considered.

### CODE REVIEW

File: .github/workflows/lint.yaml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    branches:
+      - main
+
+jobs:
+  python:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-python@v5
+      - name: Ruff checks
```
Comment (derekadombek) [inline .github/workflows/lint.yaml:18]: lol I think we both started similar type branches but I was doing it with Black. Did we decide not to use Black anymore? With black already configured in `pyproject.toml` we can just run it along side the current unit tests with poetry. Or we could probable add ruff to run in poetry instead?

File: .github/workflows/lint.yaml [hunk 2]
```diff
@@ -0,0 +1,21 @@
+name: Lint checks
+on:
+  workflow_call:
+  workflow_dispatch:
+  pull_request:
+    branches:
+      - '**'
```
Comment (derekadombek) [inline .github/workflows/lint.yaml:7]: this will need to point to the OCR directory

**Comment by derekadombek** [comment]
> @derekadombek ready for review, I think. It should only check in the ocr/ directory now. dedupe/ can opt in by specifying something in `pyproject.toml` or `ruff.toml`, though I still think we should just have one standard for the entire repository to simplify things

I personally don't mind either way how the workflows get divided. This will work but It's usually good to keep things uniform and standard across a repo and sticking with one method of doing it.

**Review by derekadombek — APPROVED** [review]

**Review by zdeveloper — APPROVED** [review]
LGTM! Thank you!
